File: research/experiments/power-thermal-probes/thermalstate_probe.m
#import <Foundation/Foundation.h>

int main(void) {
    @autoreleasepool {
        NSProcessInfoThermalState state = NSProcessInfo.processInfo.thermalState;
        printf("thermal_state_raw_value=%ld\n", (long)state);
        switch (state) {
            case NSProcessInfoThermalStateNominal: puts("thermal_state=nominal"); break;
            case NSProcessInfoThermalStateFair: puts("thermal_state=fair"); break;
            case NSProcessInfoThermalStateSerious: puts("thermal_state=serious"); break;
            case NSProcessInfoThermalStateCritical: puts("thermal_state=critical"); break;
            default: puts("thermal_state=unknown"); break;
        }
    }
    return 0;
}

File: research/experiments/power-thermal-probes/thermalstate_probe.swift
import Foundation

let processInfo = ProcessInfo.processInfo
let state = processInfo.thermalState

print("thermal_state_raw_value=\(state.rawValue)")
switch state {
case .nominal:
    print("thermal_state=nominal")
case .fair:
    print("thermal_state=fair")
case .serious:
    print("thermal_state=serious")
case .critical:
    print("thermal_state=critical")
@unknown default:
    print("thermal_state=unknown")
}
